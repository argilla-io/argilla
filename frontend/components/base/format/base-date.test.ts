import { mount } from "@vue/test-utils";
import BaseDateVue from "./BaseDate.vue";

const dateMocked = new Date("2023-07-19 00:00:00");
jest.useFakeTimers("modern").setSystemTime(dateMocked);

describe("Base Date should", () => {
  test("should format date correctly", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-19").toString(),
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 1 second ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-18 23:59:59").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 2 seconds ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-18 23:59:58").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 2 hours ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-18 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day as yesterday", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-17 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 2 days ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-16 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day last week", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-11 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 2 weeks ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-07-01 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day last month", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-06-18 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });

  test("should format day 2 months ago", () => {
    const baseDate = mount(BaseDateVue, {
      propsData: {
        date: new Date("2023-05-18 22:00").toString(),
        format: "date-relative-now",
      },
    });

    expect(baseDate).toMatchSnapshot();
  });
});
