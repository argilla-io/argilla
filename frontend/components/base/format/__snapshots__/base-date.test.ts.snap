// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Date should should format date correctly 1`] = `<span> 2023-07-19 00:00 </span>`;

exports[`Base Date should should format day 1 second ago 1`] = `<span> 1 second ago </span>`;

exports[`Base Date should should format day 2 days ago 1`] = `<span> 2 days ago </span>`;

exports[`Base Date should should format day 2 hours ago 1`] = `<span> 2 hours ago </span>`;

exports[`Base Date should should format day 2 months ago 1`] = `<span> 2 months ago </span>`;

exports[`Base Date should should format day 2 seconds ago 1`] = `<span> 2 seconds ago </span>`;

exports[`Base Date should should format day 2 weeks ago 1`] = `<span> 2 weeks ago </span>`;

exports[`Base Date should should format day as yesterday 1`] = `<span> yesterday </span>`;

exports[`Base Date should should format day last month 1`] = `<span> last month </span>`;

exports[`Base Date should should format day last week 1`] = `<span> last week </span>`;
